MotionPhotoReader/app/src/test/java/com/google/android/libraries/motionphotoreader/XmpParserTest.java

```diff
@@ -0,0 +1,44 @@
+package com.google.android.libraries.motionphotoreader;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+import com.google.common.primitives.Bytes;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class XmpParserTest {
+    private static final byte[] OPEN_ARR = "<x:xmpmeta".getBytes();  /* Start of XMP metadata tag */
+    private static final byte[] CLOSE_ARR = "</x:xmpmeta>".getBytes();  /* End of XMP metadata tag */
+
+    private String filename;
+
+    @Before
+    public void setUp() {
+        this.filename = this.getClass().getClassLoader().getResource("test_photo.jpg").getFile();
+    }
+
+    @Test
+    public void getXmpMetadata_isNotNull() throws IOException, XMPException {
+        XMPMeta meta = XmpParser.getXmpMetadata(filename);
+        assertNotNull(meta);
+    }
+
+    @Test
+    public void getXmpByteArray_hasCorrectHeader() throws IOException {
+        byte[] segArr = XmpParser.getXmpByteArray(filename);
```

Comment:
prefer if we hardcode this value instead of calculating one. You can use a hexeditor to determine the first offset (or just dump the value in logcat and replace it with hardcoded version. Reason for this is that it if our calculation is wrong in both test and implementation, we've essentially hidden the mistake and having a hard-coded value means we know the byte should start there.

Reply:
I hard-coded in another value for the Footer test, but the correct value here is just 0. Is segArr supposed to be hardcoded? I meant to test this variable (the getXmpByteArray method).